proxy/src/console/provider.rs

```diff
@@ -538,12 +545,9 @@ impl ApiLocks {
         })
     }
 
-    pub async fn garbage_collect_worker(&self, epoch: std::time::Duration) {
-        if self.permits == 0 {
-            return;
-        }
```

Comment:
Why did you remove this check?

Reply:
I guess, it's never the case that there are 0 permits and it shouldn't hurt  much.

I changed return type to `Infallible` to put it in the maintenance tasks.